lib/improver/database/spotdata_database.py

```diff
@@ -0,0 +1,276 @@
+        #                                                 [self.primary_dim] +
+        #                                                 [self.pivot_dim]]
+        for cube in self.cubelist:
+            print cube
+            for coord in self.cubelist[0].coords():
+                name = coord.standard_name or coord.long_name
+
+                if (name not in self.column_dims + [self.primary_dim] +
+                                                   [self.pivot_dim]):
+                    cube.remove_coord(coord)
+
+            # Loop over the remaining dimensions
+            for c in cube.slices_over(1):
+                cube_name = cube.name()
+                df = DataFrame(c.data,
+                               index=c.coord(self.primary_dim).points,
+                               columns=['vals'])
+
+                if self.pivot_dim:
+                    # Reshape data based on column values
+                    coords = c.coord(self.pivot_dim).points
+                    col_names = map(self.pivot_map, coords)
+                    df.insert(1, self.pivot_dim, col_names)
+                    df = df.pivot(columns=self.pivot_dim, values='vals')
+
+                if self.primary_map:
+                    # Switch the index out for a map if specified
+                    for mapping, function in zip(self.primary_map,
+                                                 self.primary_func):
+                        df.insert(0, mapping, map(function, df.index))
+
+                    # Takes significant time if a multi-index
+                    df.set_index(self.primary_map, inplace=True)
+
+                for dim, col in itertools.izip_longest(self.column_dims,
+                                                       self.column_maps):
+                    if dim in df.columns:
+                        continue
+
+                    if dim in [coord.standard_name or coord.long_name for coord in
+                                                  c.dim_coords + c.aux_coords]:
+                        coord = c.coord(dim)
+                        col_name = col or dim
+                        if len(coord.points) == 1:
+                            col_data = coord.points[0]
+                        else:
+                            col_data = coord.points
+                    else:
+                        # Should have a conditional
+                        col_name = col or dim
+                        col_data = cube.name()
+
+                    df.insert(1, col_name, col_data)
+                    if dim != self.pivot_dim:
+                        # This is rather expensive
+                        df.set_index(col_name, append=True, inplace=True)
+                try:
+                    self.df = self.df.combine_first(df)
+                except AttributeError:
+                    self.df = df
+
+        #self.df = self.df.pivot(columns = 'forecast_period', values='vals')
+
+    def constrained_to_dataframe():
```

Comment:
This is the old slower version that will be removed from this PR.

Reply:
Agreed